Sources/TensorFlow/Core/ShapedArray.swift

```diff
@@ -25,101 +24,43 @@ import CTensorFlow
 /// TensorFlow. In either mode, the buffer object owns the memory and will deallocate it on
 /// `deinit`.
 @usableFromInline
-internal final class TensorBuffer<Scalar> {
-    typealias Shape = [Int]
-
-    /// A reference type wrapping a Swift Array.
-    /// - Note: An array is used as the native storage for `TensorBuffer`. To make in-place mutation
-    ///   possible when the array is stored in an enumeration value, the array must be wrapped in a
-    ///   reference type.
-    @usableFromInline
-    final class BoxedArray {
-        var array: [Scalar]
-
-        init(_ array: __owned [Scalar]) {
-            self.array = array
-        }
-    }
-
-    enum Allocation {
-        case native(BoxedArray)
-        case tensorFlow(CTensor)
-    }
+internal class TensorBuffer<Scalar> {
+    let count : Int
 
-    let allocation: Allocation
-    let count: Int
+    init(count: Int) { self.count = count }
 
-    deinit {
-        debugLog("De-initializing tensor buffer.")
-        switch allocation {
-        case .native:
-            debugLog("Deallocating underlying buffer.")
-        case let .tensorFlow(cTensor):
-            debugLog("Deleting underlying tensor.")
-            TF_DeleteTensor(cTensor)
-        }
-        debugLog("Returning from deinit of TensorBuffer.")
+    func withUnsafeBufferPointer<R>(
+        _ body: (UnsafeBufferPointer<Scalar>) throws -> R
+    ) rethrows -> R {
+      fatalError("TensorBuffer should not be constructed directly.")
     }
 
-    init(allocation: Allocation, count: Int) {
-        self.allocation = allocation
-        self.count = count
+    func withUnsafeMutableBufferPointer<R>(
+        _ body: (inout UnsafeMutableBufferPointer<Scalar>) throws -> R
+    ) rethrows -> R {
+      fatalError("TensorBuffer should not be constructed directly.")
```

Comment:
Same as above. Error message does not match semantics of the `withUnsafeMutableBufferPointer`?